src/Microsoft.TestPlatform.AdapterUtilities/ManagedNameUtilities/ManagedNameHelper.Reflection.cs

```diff
@@ -119,11 +118,11 @@ public static void GetManagedName(MethodBase method, out string managedTypeName,
         var methodBuilder = new StringBuilder();
 
         // Namespace and Type Name (with arity designation)
+        // hierarchyPos contains [startIndexOfNamespace, endIndexOfNameSpace, endIndexOfTypeName]
         var hierarchyPos = AppendTypeString(typeBuilder, semanticType, closedType: false);
-        if (hierarchyPos is null || hierarchyPos.Length != HierarchyConstants.Levels.TotalLevelCount)
+        if (hierarchyPos is null || hierarchyPos.Length != 3)
```

Comment:
Could we avoid magic numbers? At least add some comment to explain how you get this number.

Reply:
See line 121 :)